### ISSUE TITLE
Add support for songs already on device/local songs like Spotify.

### ISSUE THREAD

**Issue by tokyocrib** [vfsfitvnm/ViMusic#57]
### Feature description

Playing and adding songs from local storage would be a great feature for the app.

### Why do you want this feature?

I will stop using Spotify if the app gets this feature.

### Additional information

![Screenshot_2022-06-25-01-13-49-185_com spotify music_1](https://user-images.githubusercontent.com/107816362/175656986-e1f40de0-908f-4b43-b62a-bb1aa22bd0f2.jpg)

**Comment by yephny** [comment]
I'm not sure if you've realized, but the app is on its way to become the best libre music player on Android. This feature is essential to hold that title, especially with directory blacklisting support. Though, I'm a little conflicted, as a part of me keeps reminding me "do one thing and do it well".

**Comment by vfsfitvnm** [comment]
Unfortunately the implementation requires too much effort. As one of the tag says, contributions are welcome :smile:

**Comment by Jak2k** [comment]
Maybe just create a entry  "localFile" in the db-items. If it isn't provided it will fallback to cache/online.

**Comment by n-ce** [comment]
someone needs to merge vimusic and [auxio](https://github.com/OxygenCobalt/Auxio)

**Comment by suva-github-625466** [comment]
yes i want also.

**Comment by Rheneas64** [comment]
Yeah support for device file is the only thing remaining for me to dump cancer YT music

**Comment by Rheneas64** [comment]
Developers should please implement this feature

**Comment by d-0l** [comment]
I read the local playlists wrong in my excitement for this app and was disappointed to discover it doesn't yet have support to play local files. If I could fill the gaps yt music has with my flacs this would be the GOAT music player.
